### ISSUE TITLE
Property 'body' does not exist on type 'Request'

### ISSUE THREAD

**Issue by devrelm** [Microsoft/TypeScript#15199]
According to the fetch standard, the [`Request` type should have a `body` on it](https://fetch.spec.whatwg.org/#requests), by virtue of it [implementing](https://fetch.spec.whatwg.org/#request-class) the [Body mixin](https://fetch.spec.whatwg.org/#body).

Technically, the actual issue here is that the [`body` property on the `Response` interface](https://github.com/Microsoft/TypeScript/blob/master/src/lib/webworker.generated.d.ts#L947) should be moved to the [`Body`](https://github.com/Microsoft/TypeScript/blob/master/src/lib/webworker.generated.d.ts#L1158) interface.

**TypeScript Version:**  2.2.2

Using the es6 lib (though all of the libs with `fetch` type definitions have this issue.)

**Code**

```ts
// A *self-contained* demonstration of the problem follows...
function getBodyFromRequest(request: Request): ReadableStream | null {
    return request.body;
}
```

**Expected behavior:**

Compiles with no errors.

**Actual behavior:**

Fails with error `Property 'body' does not exist on type 'Request'`

**Comment by rosskevin** [comment]
@devrelm why did you close this issue?  I just ran into the same thing.

**Comment by devrelm** [comment]
I had mentioned this issue in a private repo for my employer. It looks like Github closed this issue when that private-repo's issue was closed. I've had it happen once before, and I'm getting a bit upset that GitHub is being a bit _too_ helpful.

**Comment by mhegazy** [comment]
PRs welcomed. You can find more information about contributing lib.d.ts fixes at https://github.com/Microsoft/TypeScript/blob/master/CONTRIBUTING.md#contributing-libdts-fixes.

**Comment by nhhockeyplayer** [comment]
Is there a workaround for this ?
type cast some way

error TS2339: Property 'emailAddress' does not exist on type '{}'.

my koa server is not surviving the latest and greatest updates

```
    private async update(ctx: Koa.Context, next) {
        const response = await model.update(
            {
                _id: new Types.ObjectId(ctx.params.id)
            }, {
                emailAddress: ctx.request.body.emailAddress,
                password: ctx.request.body.password,
                updatedAt: new Date()
            }, {
                runValidators: true
            }
        );
        if ( response.n === 1 ) {
            ctx.body = await model.findOne({_id: new Types.ObjectId(ctx.params.id)})
        } else {
            ctx.throw(HttpStatus.NOT_FOUND,'SERVER ERROR: No entity by _id = ' + ctx.params.id + 'was found on server');
        }
    }
```

**Comment by saschanaz** [comment]
Has been fixed by https://github.com/Microsoft/TSJS-lib-generator/pull/420

**Comment by paulomendes15229** [comment]
Try use, (<any>data._body), this worked for me.

**Comment by paulomendes15229** [comment]
Try use, ( <an y>data._body), this worked for me.

**Comment by devrelm** [comment]
Closing, since this fixed by Microsoft/TSJS-lib-generator#420 (per @saschanaz's comment above)
